pkg/engine/lifecycletest/pulumi_test.go

```diff
@@ -2035,6 +2090,38 @@ func TestProviderPreviewUnknowns(t *testing.T) {
 			assert.True(t, state.DeepEquals(ins))
 		}
 
+		_, _, cstate, err := monitor.RegisterResource("pkgA:m:typB", "resB", false, deploytest.ResourceOptions{
+			Inputs: resource.PropertyMap{
+				"name": state["foo"],
+			},
+			Remote:   true,
+			Provider: provRef.String(),
+		})
+		if preview {
+			// We expect construction of remote component resources to fail during previews if the provider is
+			// configured with unknowns.
+			assert.Error(t, err)
+			assert.True(t, cstate.DeepEquals(resource.PropertyMap{}))
+		} else {
+			assert.NoError(t, err)
+			assert.True(t, cstate.DeepEquals(resource.PropertyMap{
+				"foo": resource.NewStringProperty("bar"),
+			}))
+		}
+
+		outs, _, _, err := monitor.Call("pkgA:m:typA/methodA", resource.PropertyMap{
+			"name": cstate["foo"],
+		}, provRef.String(), "")
+		assert.NoError(t, err)
+
+		if preview {
+			assert.True(t, outs.DeepEquals(resource.PropertyMap{}))
```

Comment:
I think this should probably be an error, like in Construct. 
We should probably also test Invokes here which I think should also error.